service/model/environment.go

```diff
@@ -66,32 +66,14 @@ func (e *Environment) Validate() error {
 		return errEnvironmentNameRequired
 	}
 
-	return nil
-}
-
-func ToEnvironment(id, projectID uuid.UUID, name, url string, createdAt, updatedAt time.Time) (Environment, error) {
-	u, err := toServiceURL(url)
-	if err != nil {
-		return Environment{}, err
-	}
-
-	env := Environment{
-		ID:         id,
-		ProjectID:  projectID,
-		Name:       name,
-		ServiceURL: u,
-		CreatedAt:  createdAt,
-		UpdatedAt:  updatedAt,
+	if e.ServiceURL.String() != "" && !urlx.IsAbsolute(e.ServiceURL.String()) {
+		return errEnvironmentServiceURLMustBeAbsolute
```

Comment:
I am going to update it to use validator package in the separate PR.

Reply:
Yeah, already approved. 💪